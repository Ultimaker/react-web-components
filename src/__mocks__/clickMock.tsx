// Copyright (c) 2018 Ultimaker B.V.

export const mockClickEvent = { stopPropagation: jest.fn(), preventDefault: jest.fn() };
