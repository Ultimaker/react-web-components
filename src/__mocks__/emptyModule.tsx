// Copyright (c) 2018 Ultimaker B.V.
module.exports = {};
