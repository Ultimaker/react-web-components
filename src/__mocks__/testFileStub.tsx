// Copyright (c) 2018 Ultimaker B.V.
module.exports = 'test-file-stub';