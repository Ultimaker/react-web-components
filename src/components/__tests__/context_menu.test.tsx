// Copyright (c) 2018 Ultimaker B.V.
import * as React from 'react';
import { shallow, mount } from 'enzyme';

// component
import ContextMenu from '../context_menu';

// mocks
import { mockClickEvent } from '../../__mocks__/clickMock';


describe('The ContextMenu component', () => {
    let props;
    let wrapper;

    beforeEach(() => {
        props = {
            menuWidth: 200
        };
        wrapper = shallow(<ContextMenu {...props} />);
    });

    it('should render', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render to right', () => {
        wrapper.setProps({ menuOffsetDirection: 'right' });
        expect(wrapper.find('.menu').prop('style').left).toBe(-30);
        expect(wrapper.find('.menu').prop('style').right).toBeUndefined();
    });

    it('should render to north', () => {
        wrapper.setProps({ menuDirection: 'north' });
        expect(wrapper.find('.context-menu--north')).toHaveLength(1);
        expect(wrapper.find('.context-menu--south').exists()).toBe(false);
    });

    it('should render for a panel', () => {
        wrapper.setProps({ positionMenuInPanel: true });
        expect(wrapper.find('.context-menu--panel')).toHaveLength(1);
    });

    it('should show menu when the trigger is clicked', () => {
        wrapper.find('.trigger').simulate('click', mockClickEvent);
        expect(wrapper.find('.visible')).toHaveLength(1);
    });

    it('should hide menu when the trigger is clicked', () => {
        wrapper.find('.trigger').simulate('click', mockClickEvent);
        expect(wrapper.find('.visible')).toHaveLength(1);
        wrapper.find('.trigger').simulate('click', mockClickEvent);
        expect(wrapper.find('.visible').exists()).toBe(false);
    });

    it('should hide menu on menu click', () => {
        wrapper.find('.trigger').simulate('click', mockClickEvent);
        expect(wrapper.find('.visible')).toHaveLength(1);
        wrapper.find('.container').simulate('click', mockClickEvent);
        expect(wrapper.find('.visible').exists()).toBe(false);
    });

    it('should not propagate click', () => {
        wrapper.simulate('click', mockClickEvent);
        expect(mockClickEvent.stopPropagation).toBeCalled();
    });
});
