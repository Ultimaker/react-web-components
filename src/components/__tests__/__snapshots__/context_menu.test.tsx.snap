// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The ContextMenu component should render 1`] = `
<div
  className="context-menu context-menu--south"
>
  <Button
    className="trigger"
    linkToNewTab={false}
    onClickHandler={[Function]}
    shape="rectangle"
    style="no-style"
    type="button"
  >
    <div
      style={
        Object {
          "width": 45,
        }
      }
    />
  </Button>
  <div
    className="container"
  >
    <div
      className="menu"
      style={
        Object {
          "right": -30,
          "width": 200,
        }
      }
    >
      <Collapse
        fixedHeight={-1}
        forceInitialAnimation={false}
        hasNestedCollapse={false}
        isOpened={false}
        onMeasure={[Function]}
        onRender={[Function]}
        onRest={[Function]}
        springConfig={
          Object {
            "damping": 32,
            "stiffness": 390,
          }
        }
        style={Object {}}
        theme={
          Object {
            "collapse": "ReactCollapse--collapse",
            "content": "ReactCollapse--content",
          }
        }
      >
        <ul />
      </Collapse>
    </div>
  </div>
</div>
`;
